File: đồ án giải thuật/giải thuật R/09_hang_doi-queue.R
library(dequer)
q <- queue()
for (i in 1:3) pushback(q, i)
str(q)
## queue of 3
##  $ : int 1
##  $ : int 2
##  $ : int 3
pop(q)
## [1] 1
str(q)
## queue of 2
##  $ : int 2
##  $ : int 3

File: đồ án giải thuật/giải thuật Python/09_hang_doi-queue.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jul 18 09:02:24 2021

@author: Toan
"""


# Khởi tạo hàng đợi
queue = ['a']
 
# Thêm các phần tử vào hàng đợie
queue.append('b')
queue.append('c')
queue.append('d')
 
print("Initial queue")
print(queue)
 
# Xóa các phần tử khỏi hàng đợi
print("\nElements dequeued from queue")
print(queue.pop(0))
print(queue.pop(0))

print("\nQueue after removing elements")
print(queue)
 
# bỏ print(queue.pop(0))
# sẽ tăng và IndexError
# vì hàng đợi bây giờ trống